genesis/vis/batch_renderer.py

```
@@ -84,104 +86,124 @@ def retrieve_rigid_meshes_static(self):
-
-                mat_uv_data.append(visual.uv.astype(np.float32))
-                mat_texture_widths.append(texture_width)
-                mat_texture_heights.append(texture_height)
-                mat_texture_nchans.append(texture_nchans)
-                mat_texture_offsets.append(total_texture_size)
-                mat_texture_data.append(
-                    np.asarray(visual.material.image.transpose(Image.Transpose.FLIP_TOP_BOTTOM), dtype=np.uint8).flat
-                )
-                mat_texture_ids.append(texture_ids)
-                mat_rgbas.append(visual.material.diffuse.astype(np.float32) / 255.0)
-
+            geom_surface = vgeom.surface
+            geom_textures = geom_surface.get_rgba(batch=True).textures
+
+            geom_texture_indices = []
+            for geom_texture in geom_textures:
+                if isinstance(geom_texture, gs.textures.ImageTexture) and geom_texture.image_array is not None:
+                    texture_id = geom_texture.image_path
+                    if texture_id not in texture_indices:
+                        texture_idx = num_textures
+                        if texture_id is not None:
+                            texture_indices[texture_id] = texture_idx
+                        texture_widths.append(geom_texture.image_array.shape[1])
+                        texture_heights.append(geom_texture.image_array.shape[0])
+                        assert geom_texture.channel() == 4
+                        texture_nchans.append(geom_texture.channel())
+                        texture_offsets.append(total_texture_size)
+                        texture_data.append(geom_texture.image_array.flat)
+                        num_textures += 1
+                        total_texture_size += geom_texture.image_array.size
+                    else:
+                        texture_idx = texture_indices[texture_id]
+                    geom_texture_indices.append(texture_idx)
+
+            # TODO: support batch rgba
```

What does this means? What will happen in the meantime? Should we raise an exception?